examples/context/theme-detailed-app.js

```diff
@@ -0,0 +1,30 @@
+import {ThemeContext, themes} from './theme-context';
+import ThemedButton from './button';
+
+class App extends React.Component {
+  state = {
+    theme: themes.light,
+  };
+
+  toggleTheme = () => {
+    this.setState(state => ({
+      theme:
+        state.theme === themes.dark
+          ? themes.light
+          : themes.dark,
+    }));
+  };
+
+  render() {
+    //highlight-range{2,6}
+    return (
+      <ThemeContext.Provider value={this.state.theme}>
+        <ThemedButton onClick={this.toggleTheme}>
```

Comment:
This example makes me feel like `ThemedButton` has to be a direct child of `ThemeContext.Provider`. I think it would be clearer if there was a component like

```js
function Toolbar(props) {
  return (
    <ThemedButton onClick={props.onThemeChange}>Change Theme</ThemedButton>
  );
}
```

in the middle